src/node-lib/curriculum-api-2023/queries/specialistUnitListing/specialistUnitListing.query.ts

```diff
@@ -0,0 +1,208 @@
+
+import {
+  BatchResultResponseArray,
+  DevelopmentalStage,
+  SpecialistUnitListRequestSchema,
+  SpecialistUnitListingData,
+  batchResultResponseArray,
+  developmentStageUnitCount,
+  specialistUnitLessonCount,
+  specialistUnitListRequestSchema,
+  specialistUnitListingSchema,
+} from "./specialistUnitListing.schema";
+
+import OakError from "@/errors/OakError";
+
+export const getExpandedSpecialistUnits = (
+  specialistUnits: SpecialistUnitListRequestSchema,
+  data: BatchResultResponseArray,
+) => {
+  const units = specialistUnits.map((unit, i) => {
+    const batchRes = specialistUnitLessonCount.parse(data[i]?.data);
+    if (data[i]?.data) {
+      return [
+        {
+          title: unit.unit_title,
+          slug: unit.unit_slug,
+          lessonCount: batchRes.specialistUnitLessonCount.aggregate.count,
+          nullTitle: unit.unit_title,
+          programmeSlug: unit.synthetic_programme_slug,
+          subjectSlug: unit.combined_programme_fields.subject_slug,
+          subjectTitle: unit.combined_programme_fields.subject,
+          expired: unit.expired || false,
+          expiredLessonCount:
+            batchRes.specialistUnitExpiredLessonCount.aggregate.count,
+          themeSlug: unit.combined_programme_fields.phase_slug || null,
+          themeTitle: unit.combined_programme_fields.phase || null,
+          learningThemes: [
+            {
+              themeSlug: unit.combined_programme_fields.phase_slug,
+              themeTitle: unit.combined_programme_fields.phase,
+            },
+          ],
+          developmentalStageSlug:
+            unit.combined_programme_fields.developmentstage_slug,
+          developmentalStageTitle:
+            unit.combined_programme_fields.developmentstage,
+        },
+      ];
+    }
+  });
+
+  return units;
+};
+
+export const getExpandedDevelopmentalStages = (
+  partialDevelopmentStages: Array<Partial<DevelopmentalStage>>,
+  data: BatchResultResponseArray,
+) => {
+  return partialDevelopmentStages.map((stage, i) => {
+    const batchRes = developmentStageUnitCount.parse(data[i]?.data);
+    if (data[i]?.data) {
+      return {
+        ...stage,
+        unitCount: batchRes.developmentStageUnitCount.aggregate.count,
+        lessonCount: batchRes.developmentStageLessonCount.aggregate.count,
+      };
+    }
+  });
+};
+
+export const getUnitBatchRequests = (
+  specialistUnits: SpecialistUnitListRequestSchema,
+) => {
+  return specialistUnits.map((c) => {
+    return {
+      document: SpecialistLessonCountDocument,
+      variables: { unit_slug: c.unit_slug },
+    };
+  });
+};
+
+export const getDevelopmentStagesBatchRequests = (
+  developmentalStages: Partial<DevelopmentalStage>[],
+) => {
+  return developmentalStages.map((c) => {
+    return {
+      document: DevelopmentStageUnitCountDocument,
+      variables: {
+        _contains: {
+          developmentstage_slug: c?.slug,
+        },
+        syntheticProgrammeSlug: c?.programmeSlug,
```

Comment:
I'm a bit unclear on why we need both developmentstage and programme slug for this query - doesn't the programme slug contain the developmentstage slug?